packages/shared/src/index.ts

```diff
@@ -1,34 +1,28 @@
-/**
- * Errors
+/** The following files are not exported on purpose:
+ * - cookie.ts
+ * - globs.ts
+ *
+ * The following folders are also not exported on purpose:
+ * - react
+ *
+ * People should always use @clerk/shared/<name> instead
  */
-export { buildErrorThrower } from './errors/thrower';
-export type { ErrorThrower, ErrorThrowerOptions } from './errors/thrower';
 
-export * from './errors/Error';
-
-/**
- * Utils
- */
 export * from './utils';
 
-/**
- * Hooks
- */
-export { assertContextExists, createContextAndHook } from './hooks/createContextAndHook';
-export { useOrganization } from './hooks/useOrganization';
-export { useOrganizationList } from './hooks/useOrganizationList';
-export { useOrganizations } from './hooks/useOrganizations';
-export { useSafeLayoutEffect } from './hooks/useSafeLayoutEffect';
-export {
-  ClerkInstanceContext,
-  ClientContext,
-  OrganizationContext,
-  OrganizationProvider,
-  SessionContext,
-  useClerkInstanceContext,
-  useClientContext,
-  useOrganizationContext,
-  UserContext,
-  useSessionContext,
-  useUserContext,
-} from './hooks/contexts';
+export { createWorkerTimers } from './workerTimers';
+export * from './browser';
+export * from './color';
+export * from './date';
+export * from './deprecated';
+export * from './error';
+export * from './file';
+export { handleValueOrFn } from './handleValueOrFn';
+export { isomorphicAtob } from './isomorphicAtob';
+export * from './keys';
+export { loadScript } from './loadScript';
+export { LocalStorageBroadcastChannel } from './localStorageBroadcastChannel';
+export * from './poller';
+export * from './proxy';
+export * from './underscore';
+export * from './url';
```

Comment:
(Also see the first bulletpoint in the the "Other changes" PR description)

With this PR I don't want to completely kill productivity for the sake of diminishing returns on optimizations. The purpose of the PR is to **allow** subpaths so that we (the engineers) can use them where appropriate.

If I'd remove this index file, some files would have 10+ imports without giving any actual benefit - because all these functions exported from this file here are fine to be not completely separated. They don't use any heavy libraries, they don't assume a completely different environment (like `react`). So they won't cause any problems if they might be analyzed before being tree-shaken.

Hope that makes sense :)

Reply:
With the proposed setup, is there any difference between `*` exports and named exports as far as tree-shaking is concerned?

Reply:
No, no difference. Important bit for this file is to _not_ export certain stuff to keep it separate